build/system.gyp

```diff
@@ -35,6 +35,24 @@
        ],
       ],
     },
+    {
+      'target_name': 'libnotify',
+      'type': 'none',
+      'conditions': [
+        ['building_crosswalk_bin==1', {
+          'defines': ['USE_LIBNOTIFY=1'],
```

Comment:
You need to follow what the "gio" target above does more closely: `defines` and `cflags` need to be in an `direct_dependent_settings` block, otherwise they will only be used when building this target itself.